x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/utils/helpers.tsx

```diff
@@ -870,3 +881,102 @@ export const enrichSharedExceptions = (
     });
   });
 };
+
+/**
+ * Creates new Rule exception item, fills it with
+ * new condition ANDED together from the alert highlighted fields
+ */
+export const buildRuleExceptionWithConditions = ({
+  name,
+  exceptionEntries,
+}: {
+  name: string;
+  exceptionEntries: EntriesArray;
+}): ExceptionsBuilderExceptionItem => {
+  return {
+    ...getNewExceptionItem({ listId: undefined, namespaceType: 'single', name }),
+    entries: addIdToEntries(exceptionEntries),
+  };
+};
+
+/**
+ Generate exception conditions based on the highlighted fields of the alert that 
+ have corresponding values in the alert data.
+ */
+export const buildExceptionEntriesFromAlertFields = ({
+  highlightedFields,
+  alertData,
+}: {
+  highlightedFields: EventSummaryField[];
+  alertData: AlertData;
+}): EntriesArray => {
+  return Object.values(highlightedFields).reduce((acc: EntriesArray, field) => {
+    const fieldKey = field.id;
+    const fieldValue = get(alertData, fieldKey) || get(alertData, getKibanaAlertIdField(fieldKey));
+
+    if (fieldValue)
+      acc.push({
```

Comment:
Is the field ever an array of values, or is it always a single string value? I know we're not doing nested fields for the initial implementation but wondering if we need to check for any other types?

Reply:
I'm not familiar with all the fields here, but I think it'd be worth checking if the field values are an array. [ES](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html#types-array-handling) is interesting in that any field can be an array, it must just be an array of all the same field types. If so, then it should be of type 'match_any'.

Reply:
In the case of populating a field has an array of values, for ex.  `Endpoint.capabilities`
<img width="1273" alt="image" src="https://github.com/elastic/kibana/assets/12671903/46405787-ee86-4a0b-8ac0-e91d7d665a1e">

Reply:
Played around with it together over Zoom - looks good!